sodium/crypto_sign.pony

```diff
@@ -47,7 +56,15 @@ primitive CryptoSign
     let pk_size = public_key_size(); let pk = _make_buffer(pk_size)
     if 0 != @crypto_sign_keypair[_Int](pk.cstring(), sk.cstring()) then error end
     (CryptoSignSecretKey(consume sk), CryptoSignPublicKey(consume pk))
-  
+
+  fun tag seed_keypair(seed: CryptoSignSeed): (CryptoSignSecretKey, CryptoSignPublicKey)? =>
+    let sk_size = secret_key_size(); let sk = _make_buffer(sk_size)
```

Comment:
Yeah, we'll have to see how this can be improved once we have dependent types.  Right now checking size at runtime is our only option.  But I figured that I at least provided some ways for the user to add higher-level error checking if they wish, which can see the size that the object is supposed to be and compare it to the actual size, or simply call `is_valid()` on each object.  That would give them opportunities to do error handling that knows which object had a problem, instead of having the method fail with an `error` that simply indicates "one of the passed objects was invalid".